test/gurka/test_shortcut.cc

```diff
@@ -289,3 +289,55 @@ TEST(Shortcuts, ShortcutsInBins) {
 
   EXPECT_EQ(shortcut_cnt, 1);
 }
+
+TEST(Shortcuts, ShortcutRestrictions) {
+  // in each line the center way should not be part of a shortcut due to restrictions
+  const std::string ascii_map = R"(
+  A--B--C--D--E--F
+
+  G--H--I--J--K--L
+
+  M--N--O--P--Q--R
+  )";
+  const gurka::ways ways = {
+      {"AB", {{"highway", "motorway"}, {"name", "highway"}}},
+      {"BC", {{"highway", "motorway"}, {"name", "highway"}}},
+      {"CD", {{"highway", "motorway"}, {"name", "highway"}, {"maxweight", "3.5"}}},
```

Comment:
nitpick: to be sure this is working properly we would need to also run this graph without the restrictions and prove that each line makes just 2 shortcuts instead of 4